src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -2492,8 +2515,19 @@ private int doImportKeyStoreSingle(KeyStore srckeystore, String alias)
                 CertPathConstraintsParameters cpcp =
                         buildCertPathConstraint((X509Certificate)c, null);
                 checkWeakConstraint("<" + newAlias + ">", c, cpcp);
+                keyStore.setEntry(newAlias, entry, pp);
+            } else {
+                keyStore.setEntry(newAlias, entry, pp);
+                try {
+                    Key key = keyStore.getKey(newAlias, newPass);
+                    SecretKeyConstraintsParameters skcp =
+                            new SecretKeyConstraintsParameters(key);
+                    checkWeakConstraint("<" + newAlias + ">", (SecretKey)key, skcp);
+                } catch (UnrecoverableKeyException e) {
+                    // skip
+                }
             }
-            keyStore.setEntry(newAlias, entry, pp);
+
```

Comment:
While `c == null` usually means this is a secret key entry, this is not guaranteed. How about we put the check on a secret key entry in its own `instanceof` check, then we can save a cast.

Also, the `setEntry` is always called and it can be put outside any if/else block.